src/game/logic/object/update/physicsupdate.cpp

```diff
@@ -648,3 +911,489 @@ Module *PhysicsBehavior::Friend_New_Module_Instance(Thing *thing, ModuleData con
+                            i1 = 1;
+                        }
+                    }
+                } else {
+                    i1 = 2;
+                }
+            }
+
+            float f13 = radius * 2.25f * radius;
+
+            if (i1 == 0) {
+                Coord3D pos1;
+                pos1 = *other_pos;
+                float f14 = pos1.x - pos->x;
+                float f15 = pos1.y - pos->y;
+                float f16 = f14 * dir->x + f15 * dir->y;
+                float f17 = f14 * f14 + f15 * f15;
+
+                if (f16 < 0.0f && f17 < f13) {
+                    crushed = true;
+                }
+            } else if (i1 == 2) {
+                Coord3D pos1;
+                pos1 = *other_pos;
+                pos1.x = pos1.x + x;
+                pos1.y = pos1.y + y;
+                float f14 = pos1.x - pos->x;
+                float f15 = pos1.y - pos->y;
+                float f16 = f14 * dir->x + f15 * dir->y;
+                float f17 = f14 * f14 + f15 * f15;
+
+                if (f16 < 0.0f && f17 < f13) {
+                    crushed = true;
+                }
+            } else if (i1 == 1) {
+                Coord3D pos1;
+                pos1 = *other_pos;
+                pos1.x = pos1.x - x;
+                pos1.y = pos1.y - y;
+                float f14 = pos1.x - pos->x;
+                float f15 = pos1.y - pos->y;
+                float f16 = f14 * dir->x + f15 * dir->y;
+                float f17 = f14 * f14 + f15 * f15;
+                if (f16 < 0.0f && f17 < f13) {
+                    crushed = true;
+                }
+            }
+
+            if (crushed) {
+                DamageInfo info;
+                info.m_in.m_damageType = DAMAGE_CRUSH;
+                info.m_in.m_deathType = DEATH_CRUSHED;
+                info.m_in.m_sourceID = obj->Get_ID();
+                info.m_in.m_amount = 0.0f;
+                obj->Attempt_Damage(&info);
+            }
+        }
+
+        return true;
+    } else {
+        return false;
+    }
+}
+
+void PhysicsBehavior::Add_Overlap(Object *obj)
+{
+    if (obj != nullptr) {
+        if (!Is_Currently_Overlapped(obj)) {
+            m_currentOverlap = obj->Get_ID();
+        }
+    }
+}
+
+bool PhysicsBehavior::Was_Previously_Overlapped(Object *obj) const
+{
+    return obj != nullptr && obj->Get_ID() == m_previousOverlap;
+}
+
+bool PhysicsBehavior::Is_Ignoring_Collisions_With(ObjectID id) const
+{
+    return id != OBJECT_UNK && id == m_ignoreCollisionsWith;
+}
+
+bool PhysicsBehavior::Is_Currently_Overlapped(Object *obj) const
+{
+    return obj != nullptr && obj->Get_ID() == m_currentOverlap;
+}
+
+void PhysicsBehavior::Test_Stunned_Unit_For_Destruction()
+{
+    if (Get_Flag(STUNNED)) {
+        Object *obj = Get_Object();
+        const Coord3D *pos = obj->Get_Position();
+        AIUpdateInterface *update = obj->Get_AI_Update_Interface();
+
+        if (obj->Get_Transform_Matrix()->Get_Z_Vector().Z < 0.0f || obj->Is_Outside_Map()
```

Comment:
id prefer the if elses pseudo shows here

Reply:
This is exactly how I see it in the pseudo for ZH WB.